src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/AADFedAuthTokenRefreshTest/AADFedAuthTokenRefreshTest.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections;
+using System.Linq;
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Text;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Data.SqlClient.ManualTesting.Tests
+{
+    public class AADFedAuthTokenRefreshTest
+    {
+        private readonly ITestOutputHelper _testOutputHelper;
+
+        public AADFedAuthTokenRefreshTest(ITestOutputHelper testOutputHelper)
+        {
+            _testOutputHelper = testOutputHelper;
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.IsAADPasswordConnStrSetup))]
+        public void FedAuthTokenRefreshTest()
+        {
+            // ------------------  Use settings below for local environment testing  ------------------------
+            //SqlConnectionStringBuilder builder = new(DataTestUtility.AADPasswordConnectionString);
+            //string dataSourceStr = builder.DataSource;
+
+            //// set user id and password from AADPasswordConnectionString
+            //string user = builder.UserID;
+            //string password = builder.Password;
+
+            //// Set Environment variables used for ActiveDirectoryDefault authentication type
+            //Environment.SetEnvironmentVariable("AZURE_USERNAME", $"{user}");
+            //Environment.SetEnvironmentVariable("AZURE_PASSWORD", $"{password}");
+
+            //string userEnvVar = Environment.GetEnvironmentVariable("AZURE_USERNAME");
+            //string passwordEnvVar = Environment.GetEnvironmentVariable("AZURE_PASSWORD");
+            //Assert.True($"{user}" == userEnvVar, @"AZURE_USERNAME environment variable must be set");
+            //Assert.True($"{password}" == passwordEnvVar, @"AZURE_PASSWORD environment variable must be set");
+
+            //// Local environment connection string
+            //string connStr = $"Server={dataSourceStr};Persist Security Info=False;User ID={user};MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Authentication=ActiveDirectoryDefault;Timeout=90";
+           
+            // ------------------ End of local environment settings ----------------------------------------------------
+
+            // ------------------ Pipeline environment setting ---------------------------------------------------------
+            // Use this connection string when running in a pipeline
+            string connStr = DataTestUtility.AADPasswordConnectionString;
+            // -------------------End of Pipeline environment setting --------------------------------------------------
+
+            // Create a new connection object and open it
+            SqlConnection connection = new SqlConnection(connStr);
```

Comment:
`using` - for all disposables.

Reply:
The using keyword is now in use to auto dispose connections.